collect_app/src/main/java/org/odk/collect/android/widgets/BearingWidget.java

```diff
@@ -21,176 +21,126 @@
+        textBackground = binding.answerText.getBackground();
+        binding.answerText.setBackground(null);
 
-        LinearLayout answerLayout = new LinearLayout(getContext());
-        answerLayout.setOrientation(LinearLayout.VERTICAL);
-
-        answer = getEditText();
-        textBackground = answer.getBackground();
-        answer.setBackground(null);
-
-        getBearingButton = createSimpleButton(getContext(), getFormEntryPrompt().isReadOnly(), getContext().getString(R.string.get_bearing), getAnswerFontSize(), this);
+        String answerText = questionDetails.getPrompt().getAnswerText();
+        if (answerText != null && !answerText.equals("")) {
+            binding.bearingButton.setText(getContext().getString(R.string.replace_bearing));
+            binding.answerText.setText(answerText);
+        }
+    }
 
-        answerLayout.addView(getBearingButton);
-        answerLayout.addView(answer);
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        binding = BearingWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
 
-        String s = questionDetails.getPrompt().getAnswerText();
-        if (s != null && !s.equals("")) {
-            getBearingButton.setText(getContext().getString(R.string.replace_bearing));
-            if (!isSensorAvailable) {
-                answer.setText(s);
-            }
-            setBinaryData(s);
+        if (prompt.isReadOnly()) {
+            binding.bearingButton.setVisibility(GONE);
+        } else {
+            binding.bearingButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+            binding.bearingButton.setOnClickListener(v -> onButtonClick());
         }
-        addAnswerView(answerLayout, WidgetViewUtils.getStandardMargin(context));
+        binding.answerText.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+
+        return binding.getRoot();
     }
 
     @Override
     public void clearAnswer() {
-        answer.setText(null);
-        getBearingButton.setText(getContext().getString(R.string.get_bearing));
-
+        binding.answerText.setText(null);
+        binding.bearingButton.setText(getContext().getString(R.string.get_bearing));
         widgetValueChanged();
     }
 
     @Override
     public IAnswerData getAnswer() {
-        String s = answer.getText().toString();
-
-        if (s.equals("")) {
-            return null;
-        } else {
-            return new StringData(s);
-        }
+        String answerText = binding.answerText.getText().toString();
+        return answerText.isEmpty() ? null : new StringData(answerText);
     }
 
     @Override
     public void setBinaryData(Object answer) {
-        this.answer.setText((String) answer);
+        binding.answerText.setText((String) answer);
         widgetValueChanged();
     }
 
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
         if (isSensorAvailable) {
```

Comment:
Do we really need this check? I think it's redundant.

Reply:
In case the button is disabled, the listener will not listen to any click events, so yes, it is redundant.